src/localsocketcontroller.cpp

```diff
@@ -343,9 +353,72 @@ void LocalSocketController::parseCommand(const QByteArray& command) {
   logger.warning() << "Invalid command received:" << command;
 }
 
-void LocalSocketController::write(const QJsonObject& json) {
+void LocalSocketController::write(const QJsonObject& json,
+                                  const QString& expect, int timeout) {
+  QByteArray payload = QJsonDocument(json).toJson(QJsonDocument::Compact);
```

Comment:
Should we ensure that code like DaemonAccessControl that validates commands should write empty responses for disallowed commands? Otherwise we will wrongly detect those cases as a disconnected daemon.

Reply:
That is an excellent point. The API, as designed, doesn't really have any mechanism to return an error in this scenario and that feels like a bigger oversight than I really intended to tackle in this PR. I think the behaviour has always been that when access is denied it manifests as a disconnected daemon, so nothing has really changed here.

There seems like a design question here of, how should the client behave when it is denied access?

Reply:
In particular, how to respond to the `activate()` method when access is denied seem to be more important. Without a response the client will behave as though the server is unreachable.

Reply:
Yes, it is an existing issue and I don't think anything needs to be done in this PR.